### DISCUSSION TITLE
Create group with specified ID

### DISCUSSION THREAD

**Discussion by mkorec** [keycloak/keycloak#20860] [Q&A]
Hello, I'm trying to create group with specified ID via keycloak [REST API](https://www.keycloak.org/docs-api/18.0/rest-api/index.html#_groups_resource)

There is a [GroupRepresentation ](https://www.keycloak.org/docs-api/18.0/rest-api/index.html#_grouprepresentation) with ID as optional, but if I provide my ID, I'm getting error **"Could not find child by id"**. What does it mean? I don't want to assign any child group, just provide my ID for newly created group. 

How can I create group with specified ID?

Thanks.

**Comment by danielFesenmeyer** [comment]
Short answer: You can't. The ID is autogenerated by Keycloak.

You can create a group with a certain name, and later on retrieve it by that name. If you are using subgroup, make sure that you retrieve it by its path (in order to not accidentally retrieving a group with same name on a different level).

**Reply by mkorec** [reply]
Thanks for answer.

What? How? So what is the intention of ID property in GroupRepresentation in create request?

And what about users? There is a similar issue, I can specify user ID in create request, but user is created with different ID... 

Nothing about this in docs...

**Reply by danielFesenmeyer** [reply]
The same representation is reused both for creating and reading groups, which is a pragmatic approach from my POV: You can use the same representation for reading and updating.

Some values (such as auto-generated ID) are provided in the "read" representation, but ignored on update.

Same applies to users.

**Reply by mkorec** [reply]
Makes sense, but this should be definitely mentioned in API docs.

**Reply by furkankurt1** [reply]
what about how can I add group Id's that the user have to the token? Thanks
